enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,278 @@
+spec and status for each platform. They are added to the spec so that users can edit the
+spec day 2 in order to update the default storage class. The status field would
+be updated by the cluster-storage operator/CSI driver.
+
+#### AWS 
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+	// [... existing fields ...]
+
+    // The KMS key to be used in the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+
+// AWSPlatformStatus holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformStatus struct {
+    // [... existing fields ...]
+
+    // The KMS key used by the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+```
+
+#### Azure
+
+```go
+// AzureDiskEncryptionSet defines the configuration for a disk encryption set.
+type AzureDiskEncryptionSet struct {
+  // SubscriptionID defines the Azure subscription that contains the disk encryption set.
+  SubscriptionID string `json:"subscriptionId"`
+  
+  // ResourceGroup defines the Azure resource group that contains the disk encryption set.
+  ResourceGroup string `json:"resourceGroup"`
+  
+  // Name is the name of the disk encryption set.
+  Name string `json:"name"`
+}
+
+// AzurePlatformSpec holds the desired state of the Azure infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AzurePlatformSpec struct{
+    // DiskEncryptionSet describes an existing disk encryption set containing
+	// user-managed encryption keys to be used by the default storage class to
+	// encrypt volumes.
+    // +optional
+    DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+
+// AzurePlatformStatus holds the current status of the Azure infrastructure provider.
+type AzurePlatformStatus struct {
+    // [... existing fields ...]
+
+    // DiskEncryptionSet contains the user-managed keys used by the default
+	// storage class to encrypt volumes.
+    // +optional
+    DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+```
+
+#### GCP
+
+```go
+// KMSKeyReference gathers required fields for looking up a GCP KMS Key
+type GCPKMSKeyReference struct {
+  // Name is the name of the customer managed encryption key to be used for the disk encryption.
+  Name string `json:"name"`
+  
+  // KeyRing is the name of the KMS Key Ring which the KMS Key belongs to.
+  KeyRing string `json:"keyRing"`
+  
+  // ProjectID is the ID of the Project in which the KMS Key Ring exists.
+  // Defaults to the VM ProjectID if not set.
+  // +optional
+  ProjectID string `json:"projectID,omitempty"`
+  
+  // Location is the GCP location in which the Key Ring exists.
+  Location string `json:"location"`
+}
```

Comment:
Same question: the StorageClass parameter should be a fully qualified resource name, would it make sense to have a single field like we do in AWS?

Reply:
xref: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/6e3972c50121fd6a21f4aea85cabe59286898b51/README.md#createvolume-parameters 

and 

https://cloud.google.com/kubernetes-engine/docs/how-to/using-cmek#create_a_cmek_protected_attached_disk

Reply:
@bertinatto the only thing I feel strongly about is that I do not think we should duplicate the values. So I think the API should either be structured like it is now **OR** a single string. 

Like I said regarding Azure, I don't feel strongly about this, but with AWS already using an ARN, perhaps it is "more consistent" to use the single string. Plus I imagine that is more convenient for the consumers to not need to construct the string.

I will update the enhancement so that Azure and GCP use a single string.